### DISCUSSION TITLE
How to integrate the new cpp file with Pytorch geometric?

### DISCUSSION THREAD

**Discussion by shivanisankhyan** [pyg-team/pytorch_geometric#7736] [Q&A]
I am using neighbour loader function in my code, which uses sample_adj_cpu function to sample neighbours.Instead of random sampling i am trying to do probabilistic sampling so I am making some changes in this function which is present in the following file.

File link:
[[pytorch_sparse](https://github.com/rusty1s/pytorch_sparse/tree/master)/[csrc](https://github.com/rusty1s/pytorch_sparse/tree/master/csrc)/[cpu](https://github.com/rusty1s/pytorch_sparse/tree/master/csrc/cpu)
/sample_cpu.cpp](url)

How to integrate these changes in Pytorch geometric?

**Comment by akihironitta** [comment]
@shivanisankhyan I think you just need to rebuild the package like this:
https://github.com/rusty1s/pytorch_sparse/blob/40693abc289d155da660ec8f7c50d7edfb44e526/.github/workflows/building.yml#L89-L102

**Comment by shivanisankhyan** [comment]
I want to do this but i need help in rebuilding the package.
@rusty1s @akihironitta

**Comment by LukeLIN-web** [comment]
I believe recompile the project is enough
`pip uninstall pytorch-geometric` 
then 
`pip install . `

**Comment by shivanisankhyan** [comment]
How pip install . will install the pytorch libraries?
@LukeLIN-web

**Reply by LukeLIN-web** [reply]
I don't think so. PyG will revoke pytorch libraries. You can rebuild and reinstall pytorch independently if necessary.

**Reply by akihironitta** [reply]
@shivanisankhyan Let me know via PyG Slack if you still need help. I'm happy to help to improve PyG dev experience ;)
